### PR TITLE
Feature/workflow steps alt conditionals fix

### PR DESCRIPTION
**Pull request by LosBombz**
Updates to the workflow executions that change how workflow steps are executed and how conditionals are evaluated. The code is pretty rough so open to all suggestions for improvement/cleanup

### CODE REVIEW

File: app/mastadon/mastadon.component.js [hunk 1]
```diff
@@ -69,6 +69,139 @@ Component.createWorkflowStep = function(name, dependencies, action) {
...
+			stepNames = list.getStepNames();
+			conditionEvaluationIndex = getIndexOfLastFulfilledConditionDependency(stepNames, context, conditionalDependencies);
+		}
+
+	let executeStep = () => {
+		if (!currentStep) {
+			return;
+		}
+
+		if(conditionalDependencies) {
```
Comment (flyiniggle) [inline app/mastadon/mastadon.component.js]: quick poll: would "conditionDependencies" make more sense than "conditionalDependencies"? I can see how the latter could be misleading.

File: app/mastadon/mastadon.component.js [hunk 2]
```diff
@@ -69,6 +69,139 @@ Component.createWorkflowStep = function(name, dependencies, action) {
...
+			if(currentIndex === conditionEvaluationIndex) {
```
Comment (LosBombz) [inline app/mastadon/mastadon.component.js]: ah yes, that sounds good

File: app/mastadon/mastadon.component.js [hunk 3]
```diff
@@ -69,6 +69,139 @@ Component.createWorkflowStep = function(name, dependencies, action) {
...
+			currentStep = nextStep;
+			if(currentStep){
+				nextStep = currentStep.next;
+			}
+			currentIndex += 1;
+			executeStep();
+		}
+	}
+
+	executeStep();
```
Comment (LosBombz) [inline app/mastadon/mastadon.component.js]: worth a try

File: app/mastadon/mastadon.component.js [hunk 4]
```diff
@@ -69,6 +69,139 @@ Component.createWorkflowStep = function(name, dependencies, action) {
...
+};
```
Comment (LosBombz) [inline app/mastadon/mastadon.component.js]: awesome, appreciate it.

File: app/mastadon/mastadon.component.js [hunk 5]
```diff
@@ -69,6 +69,139 @@ Component.createWorkflowStep = function(name, dependencies, action) {
...
+	});
+};
+
+// returns true if any of the conditions have evaluated to true
+const checkConditionsMet = function(evaluatedConditions) {
+	return !!(evaluatedConditions.indexOf(true) !== -1);
+};
+
+
+const getUnfulfilledConditionDependencies = function(steps, startingContext, dependencies) {
```
Comment (LosBombz) [inline app/mastadon/mastadon.component.js]: ah ok I can remove then

Comment (flyiniggle) [inline app/mastadon/mastadon.component.js]: no, i meant more like we should put the error throwing back in, rather than taking this out :)

**Comment by LosBombz** [comment]
There are some good notes on enhancements / cleanup here, I'll document them for possible future enhancement

**Pull request merged**